CMake: ARCH roots issue

**Describe the bug**
I'm not able to build since https://github.com/zephyrproject-rtos/zephyr/pull/41301.
This is likely due to my local installation, but I can't get what should be updated.

For instance, when running `west build -p auto -b nucleo_f401re samples/hello_world/`, I'm getting the following error:
```
CMake Error at /local/mcu/zephyrproject/zephyr/cmake/arch.cmake:41 (message):
  Could not find ARCH=nucleo_f401re for BOARD=nucleo_f401re, please check
  your installation.  ARCH roots searched:
```
Issue shows up with with https://github.com/zephyrproject-rtos/zephyr/commit/254e35d61d59feca6e9b91fcc9ce7d3416917c22

I've tried to re-run `source zephyr-env.sh` with no success

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04.6 LTS
 - Toolchain (e.g Zephyr SDK, ...): zephyr-sdk-0.13.2
 - CMake: 3.22.1
 - Python 3.8

**Expected behavior**
A clear and concise description of what you expected to happen.

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)

**Logs and console output**
Here is the full log to the error:
```
$ west -vvv build -p auto -b nucleo_f401re samples/hello_world/
args namespace: Namespace(command='build', help=None, verbose=3, zephyr_base=None)
ZEPHYR_BASE=/local/mcu/zephyrproject/zephyr (origin: env)
args: Namespace(board='nucleo_f401re', build_dir=None, build_opt=[], cmake=False, cmake_only=False, command='build', dry_run=False, force=False, help=None, pristine='auto', source_dir=None, target=None, test_item=None, verbose=3, zephyr_base=None) remainder: ['samples/hello_world/']
source_dir: samples/hello_world/ cmake_opts: None
setting up build directory
setting up source directory
config dir-fmt: build
build is a zephyr build directory
build dir: build
pristine: auto auto_pristine: True
/local/mcu/zephyrproject/zephyr/build is a zephyr build directory
setting up source directory
sanity checking the build
/local/mcu/zephyrproject/zephyr/samples/hello_world is NOT a valid zephyr build directory
APPLICATION_SOURCE_DIR: /local/mcu/zephyrproject/zephyr/samples/hello_world
pristine: True
CACHED_BOARD: nucleo_f401re
-- west build: generating a build system
cmake version 3.22.1 is OK; minimum version is 3.13.1
Running CMake: /local/home/frq07517/.local/bin/cmake -DWEST_PYTHON=/usr/bin/python3.8 -B/local/mcu/zephyrproject/zephyr/build -S/local/mcu/zephyrproject/zephyr/samples/hello_world -GNinja
Including boilerplate (Zephyr base (cached)): /local/mcu/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /local/mcu/zephyrproject/zephyr/samples/hello_world
-- Zephyr version: 3.0.99 (/local/mcu/zephyrproject/zephyr)
-- Found west (found suitable version "0.12.0", minimum required is "0.7.1")
-- Board: nucleo_f401re
CMake Error at /local/mcu/zephyrproject/zephyr/cmake/arch.cmake:41 (message):
  Could not find ARCH=nucleo_f401re for BOARD=nucleo_f401re, please check
  your installation.  ARCH roots searched:

  /local/mcu/zephyrproject/zephyr
Call Stack (most recent call first):
  /local/mcu/zephyrproject/zephyr/cmake/app/boilerplate.cmake:387 (include)
  /local/mcu/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /local/mcu/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  CMakeLists.txt:5 (find_package)

```

--- a/cmake/modules/zephyr_default.cmake
+++ b/cmake/modules/zephyr_default.cmake
@@ -31,6 +31,21 @@ find_package(ZephyrBuildConfiguration
   NO_CMAKE_SYSTEM_PACKAGE_REGISTRY
 )
 
+# Test and error-out if we are affected by the PyPI CMake 3.22.1 / 3.22.2 bug
+if(${CMAKE_VERSION} VERSION_EQUAL 3.22.1 OR ${CMAKE_VERSION} VERSION_EQUAL 3.22.2)
+  # It seems only pip-installed builds are affected so we test to see if we are affected
+  cmake_path(GET ZEPHYR_BASE PARENT_PATH test_cmake_path)
+  if(ZEPHYR_BASE STREQUAL test_cmake_path)
+    message(FATAL_ERROR "The CMake version ${CMAKE_VERSION} installed suffers"
+            " the \n 'cmake_path(... PARENT_PATH)' bug, see: \n"
+	    "https://gitlab.kitware.com/cmake/cmake/-/issues/23187\n"
+	    "https://github.com/scikit-build/cmake-python-distributions/issues/221\n"
+	    "Please install another CMake version or use a build of CMake that"
+	    " does not come from PyPI."
+    )
+  endif()
+endif()
+
 # Prepare user cache
 list(APPEND zephyr_cmake_modules python)
 list(APPEND zephyr_cmake_modules user_cache)

